### PR TITLE
332 handle non initialized image embeddings better in the plugins

### PR DESCRIPTION
**Pull request by lufre1**

### CODE REVIEW

File: micro_sam/sam_annotator/_widgets.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import z5py
 
 from qtpy import QtWidgets
+from qtpy.QtWidgets import QApplication, QDialog, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QMessageBox
```
Comment (constantinpape) [inline micro_sam/sam_annotator/_widgets.py:19]: Let's not duplicate these imports. We import `qtpy.QtWidgets` in the line above, so you can use `QtWidgets.QApplication` etc. throughout the code.

File: micro_sam/sam_annotator/_widgets.py [hunk 2]
```diff
@@ -171,6 +172,35 @@ class PBarSignals(QObject):
...
+        cancel_button = QPushButton("Cancel")
+        cancel_button.clicked.connect(lambda: self.button_clicked(cancel_button))  # Connect to clicked signal
+        button_box.addWidget(cancel_button)
+
+        layout.addLayout(button_box)
+        self.setLayout(layout)
+
+    def button_clicked(self, button):
+        if button.text() == "OK":
+            self.accept()  # Accept the dialog
```
Comment (constantinpape) [inline micro_sam/sam_annotator/_widgets.py:199]: Where are `self.accept` and `self.reject` defined? Is this already implemented in `QDialog`?

File: micro_sam/sam_annotator/_widgets.py [hunk 3]
```diff
@@ -655,7 +681,76 @@ def _create_settings_widget(self):
...
+                                return {
+                                    "message_type": "error",
+                                    "message": f"The embeddings don't match with the image: {img_signature} {f.attrs['data_signature']}"
+                                }
+
+                        # Load existing parameters
+                        if "tile_shape" in f.attrs:
+                            self.tile_x, self.tile_y = f.attrs["tile_shape"]
+                            self.halo_x, self.halo_y = f.attrs["halo"]
+                            self.model_type = f.attrs["model_type"]
```
Comment (constantinpape) [inline micro_sam/sam_annotator/_widgets.py:708]: We also need to update these values in the GUI here. See https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/sam_annotator/util.py#L671

Comment (constantinpape) [reply micro_sam/sam_annotator/_widgets.py:708]: Nvm, you do this below.

File: micro_sam/sam_annotator/_widgets.py [hunk 4]
```diff
@@ -655,7 +681,76 @@ def _create_settings_widget(self):
...
+            self.user_cancelled = True
+        elif message_type == "info":
+            info_dialog = InfoDialog(title="Validation Message", message=message)
+            result = info_dialog.exec_()
+            if result == QDialog.Rejected:  # Check for cancel
+                self.user_cancelled = True  # Set flag directly in calling function
+
     def __call__(self):
+        # validate user inputs
+        self.user_cancelled = False  # Flag to track cancellation
```
Comment (constantinpape) [inline micro_sam/sam_annotator/_widgets.py:737]: We don't need to make this a class variable.

File: micro_sam/sam_annotator/_widgets.py [hunk 5]
```diff
@@ -655,7 +681,76 @@ def _create_settings_widget(self):
...
+        validation_result = self._validate_inputs()
+        if validation_result is not None:
+            self.generate_message(validation_result["message_type"], validation_result["message"])
```
Comment (constantinpape) [inline micro_sam/sam_annotator/_widgets.py:740]: Just return `user_cancelled` here.

File: micro_sam/sam_annotator/_widgets.py [hunk 6]
```diff
@@ -171,6 +172,35 @@ class PBarSignals(QObject):
     pbar_reset = Signal()
 
 
+class InfoDialog(QDialog):
+    def __init__(self, title, message):
+        super().__init__()
+        self.setWindowTitle(title)
+        self.user_cancelled = False  # Initialize flag
```
Comment (constantinpape) [inline micro_sam/sam_annotator/_widgets.py:179]: Is this used here?

**Review by constantinpape — COMMENTED** [review]
Looks good overall, some minor comments.

File: micro_sam/sam_annotator/_widgets.py (hunk 2 above)
Comment (lufre1) [reply micro_sam/sam_annotator/_widgets.py:199]: yes, it is already implemented in QDialog - https://doc.qt.io/qt-6/qdialog.html#accept

File: micro_sam/sam_annotator/_widgets.py (hunk 6 above)
Comment (lufre1) [reply micro_sam/sam_annotator/_widgets.py:179]: removed `user_cancelled`

**Comment by lufre1** [comment]
Now, the handling of embeddings missing entirely should be added

**Review by constantinpape — APPROVED** [review]

**Comment by lufre1** [comment]
> I had a quick look and this looks good from my end now. Let me know if you want to add anything else @lufre1, otherwise I will go ahead and merge it.

pushed some tiny changes, should be ready to merge from my side

**Comment by constantinpape** [comment]
Thanks @lufre1 and looks good. I will merge it later after all tests have passed.

**Pull request merged**